src/NuGet.Core/NuGet.Protocol.Core.v3/PackageUpdateResource.cs

```diff
@@ -0,0 +1,601 @@
+            }
+            else
+            {
+                log.LogInformation(Strings.DeleteCommandCanceled);
+            }
+        }
+
+        private async Task PushSymbols(string packagePath, 
+            string apiKey, 
+            ILogger log, 
+            CancellationToken token)
+        {
+            // Get the symbol package for this package
+            string symbolPackagePath = GetSymbolsPath(packagePath);
+
+            // Push the symbols package if it exists
+            if (File.Exists(symbolPackagePath))
+            {
+                // See if the api key exists
+                if (String.IsNullOrEmpty(apiKey))
+                {
+                    log.LogWarning(string.Format(CultureInfo.CurrentCulture,
+                        Strings.Warning_SymbolServerNotConfigured,
+                        Path.GetFileName(symbolPackagePath),
+                        Strings.DefaultSymbolServer));
+                }
+                string source = NuGetConstants.DefaultSymbolServerUrl;
+                await PushPackage(symbolPackagePath, source, apiKey, log, token);
+            }
+        }
+
+        /// <summary>
+        /// Get the symbols package from the original package. Removes the .nupkg and adds .symbols.nupkg
+        /// </summary>
+        private static string GetSymbolsPath(string packagePath)
+        {
+            string symbolPath = Path.GetFileNameWithoutExtension(packagePath) + NuGetConstants.SymbolsExtension;
+            string packageDir = Path.GetDirectoryName(packagePath);
+            return Path.Combine(packageDir, symbolPath);
+        }
+
+
+        private async Task PushPackage(string packagePath, 
+            string source, 
+            string apiKey, 
+            ILogger log, 
+            CancellationToken token)
+        {
+            if (string.IsNullOrEmpty(apiKey) && !IsFileSource(source))
+            {
+                log.LogWarning(string.Format(CultureInfo.CurrentCulture,
+                    Strings.NoApiKeyFound,
+                    GetSourceDisplayName(source)));
+            }
+
+            IEnumerable<string> packagesToPush = GetPackagesToPush(packagePath);
+
+            EnsurePackageFileExists(packagePath, packagesToPush);
+            foreach (string packageToPush in packagesToPush)
+            {
+                await PushPackageCore(source, apiKey, packageToPush, log, token);
+            }
+        }
+
+        private async Task PushPackageCore(string source,
+            string apiKey,
+            string packageToPush,
+            ILogger log,
+            CancellationToken token)
+        {
+            var sourceUri = new Uri(source);
+            string sourceName = GetSourceDisplayName(source);
+
+            log.LogInformation(string.Format(CultureInfo.CurrentCulture,
+                Strings.PushCommandPushingPackage,
+                Path.GetFileName(packageToPush),
+                sourceName));
+
+            if (sourceUri.IsFile)
```

Comment:
i will leverage the "IsHttp" check we already did at resource provider level, which means the "_httpSource" field is null.